tests/functional/src/main/java/org/mule/functional/classloading/isolation/maven/DependencyGraphMavenDependenciesResolver.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.functional.classloading.isolation.maven;
+
+import static java.nio.file.Files.readAllLines;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.BasicFileAttributeView;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Uses maven https://github.com/ferstl/depgraph-maven-plugin plugin to resolve the dependencies for the test class.
+ * It relies on the dot graph generated by the depgraph-maven-plugin and the artifact should have set this plugin
```

Comment:
Add a reference to DEPENDENCIES_GRAPH_FILE_NAME instead of using "the dot graph"